### DISCUSSION TITLE
Use variables in recordsList

### DISCUSSION THREAD

**Discussion by bartlomiejwtb** [YetiForceCompany/YetiForceCRM#17587] [How to] [answered]
Hello,
I have a question regarding customizing recordsList with use of variables. In my case I want to send email (with use of email template) to each employee with list of in progress project tasks assigned to them. Allow me to explain my problem first.

**Without use of the variables:**

In my email template if I write user ID statically like this:

![1](https://github.com/user-attachments/assets/51591695-4a0a-4212-9bb4-dea895effc45)

then the records list is created correctly, this is the output email content:

![2](https://github.com/user-attachments/assets/db0c07fb-b4de-497f-901b-3205d7eed82b)

**With use of the variables:**

Instead of static user ID I want to use variable that points on record owner ID like this:

![3](https://github.com/user-attachments/assets/60df1504-beec-44a0-ab7c-654905c41098)

But in the output, list is not created, although ID was interpreted correctly:

![4](https://github.com/user-attachments/assets/7f3f07d3-1285-48b8-adef-a7ae5a95982d)

**My question is simply: how to make variables work in recordsList?**

YetiForce version: 6.3.0
Database: MySQL
PHP version: 7.3.x

**Comment by mariuszkrzaczkowski** [comment]
I think it was changed in the latest version.

What you need can be simpler, create a filter and give the conditions there and just use the filter in the template

**Comment by rskrzypczak** [comment]
@bartlomiejwtb 
Unfortunately you can't set a variable within a variable, it won't work, as you yourself noticed.
.. but you have a few other options

- use twig (but it only works in PDF)
  https://github.com/YetiForceCompany/YetiForceCRM/discussions/16588#discussioncomment-3892560
- use workflow e-mail raport
  ![reportWoorkflow](https://github.com/user-attachments/assets/212a5353-5678-4c32-9341-98e118393660)
  and in email template set 
  `$(custom : UserRecordsList|__MODULE_NAME__|__FIELDS_TO_SHOW__|__CONDITIONS__|__ORDER_BY__|__LIMIT__|__VIEW_ID__|__ADVANCE_CONDITIONS__)$`
  Example: `$(custom : UserRecordsList|SSalesProcesses|subject:related_to|[[["description","y",""]]]|[["createdtime","ASC"]]|50)$`

It's not exactly what you're describing, but it might give you more scope and meet your requirements in some way

**Comment by bartlomiejwtb** [comment, marked as answer]
@mariuszkrzaczkowski @rskrzypczak 
Thank you very much for your suggestions. At the end of the day I couldn't implement your solutions in my system (1. I don't understand how to put filter in email template 2. Can't use twig since I want to print content in email, not in pdf 3. I can't see "Email report" task like on your gif).

However, with help of your hints I managed to solve my problem in kinda primitive way:

Since number of employees in our department is relatively low I decided to create **workflow** and **email template** for each employee.

1. I created workflows for each employee that project tasks lists will be sent to (trigger every workday at 7:45 am)

![Zrzut ekranu 2024-09-06 152827](https://github.com/user-attachments/assets/24dd5c1e-ed70-4803-be3c-e4496bc030d4)

2. I only want to send email to specific employee so in each workflow I set condition that will check if it's the correct employee by his name and last name

![Zrzut ekranu 2024-09-06 153055](https://github.com/user-attachments/assets/42e8a1a3-b7eb-4132-a54e-059e39ddd81a)

3. I create email template for each employee. The only thing that changes is user id at the end of you records list and name of employee in title.

![Zrzut ekranu 2024-09-06 153224](https://github.com/user-attachments/assets/5da57c84-7daf-49de-aae9-4cfdb932e103)

4. In workflow task I choose send email from template that I just created and set email from "Company email" field, or you can type it statically.

![Zrzut ekranu 2024-09-06 153122](https://github.com/user-attachments/assets/a6d1c152-3a7a-47b1-995d-08ae28b48294)

And that's how I made it work. Once again, thank your for your contribution.
